bridge-fungible-token/src/bridge_fungible_token.sw

```diff
@@ -0,0 +1,151 @@
+contract;
+
+dep data;
+dep errors;
+dep events;
+dep utils;
+
+use bridge_fungible_token_abi::BridgeFungibleToken;
+use contract_message_receiver::MessageReceiver;
+use core::num::*;
+use errors::BridgeFungibleTokenError;
+use events::{RefundRegisteredEvent, WithdrawalEvent};
+use std::{
+    chain::auth::{
+        AuthError,
+        msg_sender,
+    },
+    constants::ZERO_B256,
+    context::{
+        call_frames::{
+            contract_id,
+            msg_asset_id,
+        },
+        msg_amount,
+    },
+    logging::log,
+    storage::StorageMap,
+    vm::evm::evm_address::EvmAddress,
+};
+use utils::{
+    burn_tokens,
+    correct_input_type,
+    decompose,
+    input_message_data,
+    input_message_data_length,
+    input_message_recipient,
+    input_message_sender,
+    mint_and_transfer_tokens,
+    parse_message_data,
+    safe_b256_to_u64,
+    send_message_output,
+    transfer_tokens,
+    u64_to_b256,
+};
+
+////////////////////////////////////////
+// Storage declarations
+////////////////////////////////////////
+storage {
+    refund_amounts: StorageMap<(EvmAddress, EvmAddress), Option<b256>> = StorageMap {},
```

Comment:
consider using only b256 values in storage, rather than `EvmAddress`'s ( the SDK doesn't have this type yet anyway...)

Reply:
that might be a good idea if we want to use this contract with other bridged chains too